projects/06/Assembler/src/main/java/org/project06/Assembler.java

```diff
@@ -16,74 +16,77 @@ public static void main(String[] args) {
             System.out.println("Usage: java Assembler filename.asm");
             System.exit(1);
         }
-
         String fileName = args[0];
         String fileNameNoExt = getFilenameWithoutExtension(fileName);
 
-        // First pass write labels to hash table
-        SymbolTable symbolTable = new SymbolTable();
-        try {
-            Parser parser = new Parser(fileName);
-            int instructionCounter = 0;
-            while (parser.hasMoreCommands()) {
-                String command = parser.advance();
-                if ("L_COMMAND".equals(parser.identifyCommandType(command))) {
-                    symbolTable.addEntry(parser.retrieveSymbolOfCommand("L_COMMAND", command), instructionCounter);
-                } else {
-                    instructionCounter++;
-                }
-            }
-            parser.close();
-        } catch (FileNotFoundException e) {
-            throw new RuntimeException(e);
-        }
-
-        // Second pass handle symbols and write
         try (FileWriter writer = new FileWriter(fileNameNoExt + OUTPUT_FILE_EXTENSION);
              BufferedWriter bufferedWriter = new BufferedWriter(writer)) {
 
-            Parser parser = new Parser(fileName);
-            Code code = new Code();
+            // First pass write labels to hash table
+            SymbolTable symbolTable = constructSymbolTable(fileName);
```

Comment:
I like this refactor. Main was quite huge... Now is more clear and the flow is self explanatory. 👍